### PR TITLE
[BugFix]: Fix Bagel generate_image denoising schedule

### LINKED ISSUE #4470: [Bug]: Bagel `generate_image` denoising schedule diverges from official BAGEL in v0.22.0+
## Summary

Starting in **v0.22.0**, `Bagel.generate_image` in vLLM-Omni uses a denoising timestep schedule that does **not** match the official [ByteDance-Seed/BAGEL](https://github.com/ByteDance-Seed/BAGEL) reference implementation. The sigma/timestep **shift formula** is unchanged, but the **number of denoise iterations** is off by one for the same `num_timesteps` / `num_inference_steps` setting.

v0.20.0 matches official BAGEL. v0.22.0+ (including current `main`) diverged when Lance support was merged.

## Expected behavior

Official BAGEL (`modeling/bagel/bagel.py`, `generate_image`) builds the flow-matching schedule as:

```python
timesteps = torch.linspace(1, 0, num_timesteps, device=x_t.device)
timesteps = timestep_shift * timesteps / (1 + (timestep_shift - 1) * timesteps)
dts = timesteps[:-1] - timesteps[1:]
timesteps = timesteps[:-1]
```

Reference: [ByteDance-Seed/BAGEL `modeling/bagel/bagel.py` (main)](https://github.com/ByteDance-Seed/BAGEL/blob/main/modeling/bagel/bagel.py#L693-L696)

With `num_timesteps=50` (the default in [official `inferencer.py`](https://github.com/ByteDance-Seed/BAGEL/blob/main/inferencer.py)), this yields **49** denoise loop iterations.

vLLM-Omni **v0.20.0** uses the same schedule:
- [v0.20.0 `bagel_transformer.py`](https://github.com/vllm-project/vllm-omni/blob/v0.20.0/vllm_omni/diffusion/models/bagel/bagel_transformer.py)

## Actual behavior

Since **v0.22.0**, vLLM-Omni `Bagel.generate_image` uses:

```python
timesteps = torch.linspace(1, 0, num_timesteps + 1, device=x_t.device)
timesteps = timestep_shift * timesteps / (1 + (timestep_shift - 1) * timesteps)
dts = timesteps[:-1] - timesteps[1:]
timesteps = timesteps[:-1]
```

Reference: [v0.22.0 `bagel_transformer.py`](https://github.com/vllm-project/vllm-omni/blob/v0.22.0/vllm_omni/diffusion/models/bagel/bagel_transformer.py#L1697-L1703) (same on current `main`)

With `num_timesteps=50`, this yields **50** denoise loop iterations — one more than official BAGEL.

The in-code comment incorrectly states that both Lance and BAGEL upstream use the `num_timesteps + 1` convention. That is true for [ByteDance Lance](https://github.com/bytedance/Lance/blob/main/modeling/lance/lance.py) but **not** for official BAGEL.

## Regression timeline

| Version | `linspace` call | Denoise steps (`num_timesteps=50`) | Matches official BAGEL? |
|---------|-----------------|----------------------------------|-------------------------|
| v0.20.0 | `linspace(1, 0, num_timesteps)` | 49 | Yes |
| v0.22.0+ | `linspace(1, 0, num_timesteps + 1)` | 50 | No |

Introduced in PR #3710 ([New Model] Add Lance, commit `b76291d6`), which aligned `Bagel.generate_image` with Lance's schedule instead of keeping BAGEL's.

The unit test was also updated to encode the Lance convention:
- v0.20.0: `EXPECTED_STEPS = NUM_TIMESTEPS - 1`
- v0.22.0+: `EXPECTED_STEPS = NUM_TIMESTEPS`

## What is *not* affected

The following are consistent between official BAGEL and vLLM-Omni v0.22.0+:

- Timestep shift (sigma remapping): `σ' = shift * σ / (1 + (shift - 1) * σ)`
- ODE update: `x_t = x_t - v_t * dt`
- Default hyperparameters (`timestep_shift=3.0`, `cfg_interval=(0.4, 1.0)`) per [BAGEL README](https://github.com/ByteDance-Seed/BAGEL#-notice)

Only the **discretization count** differs.

## Impact

1. **Inference parity**: For the same `num_inference_steps`, vLLM-Omni v0.22.0+ Bagel produces different outputs than official BAGEL (and v0.20.0), because the denoising trajectory uses a different number of Euler steps over a slightly different sigma grid.

2. **RL / trajectory-based training**: Downstream projects that reconstruct sigma schedules from official BAGEL semantics (e.g. RL rollouts, log-prob computation over denoising trajectories) must special-case vLLM-Omni ≥ 0.22.0. A mismatch between rollout sampling and reference BAGEL behavior can silently degrade training.

3. **Misleading documentation in code**: The comment claiming BAGEL upstream uses `num_timesteps + 1` is incorrect and may cause further divergence in future contributions.

## Steps to reproduce

1. Set `num_timesteps = 50`, `timestep_shift = 3.0`.
2. Compare schedule construction:

```python
import torch

def steps(n, plus_one=False):
    pts = n + 1 if plus_one else n
    ts = torch.linspace(1, 0, pts)
    ts = 3.0 * ts / (1 + 2.0 * ts)
    return len(ts[:-1])

assert steps(50, plus_one=False) == 49  # official BAGEL / v0.20.0
assert steps(50, plus_one=True) == 50  # v0.22.0+
```

3. Run text-to-image with identical seeds and hyperparameters on official BAGEL vs vLLM-Omni v0.22.0+; outputs will differ due to the extra denoise step.

## Suggested fix

Restore official BAGEL schedule in `Bagel.generate_image`:

```python
timesteps = torch.linspace(1, 0, num_timesteps, device=x_t.device)
```

Keep the `num_timesteps + 1` convention only in the Lance code path (`LanceBagel` / Lance pipeline), where upstream Lance explicitly uses it.

Also update:
- The misleading comment in `bagel_transformer.py`
- `tests/diffusion/models/bagel/test_trajectory_recording.py` to expect `NUM_TIMESTEPS - 1` steps again for Bagel

## Environment

- vLLM-Omni: v0.22.0 (also reproducible on current `main`)
- Model: BAGEL-7B-MoT
- Reference: [ByteDance-Seed/BAGEL](https://github.com/ByteDance-Seed/BAGEL) `main`

### Your code version

.

### 🐛 Describe the bug

.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by princepride**
## Purpose
Fix #4470

### CODE REVIEW

File: vllm_omni/diffusion/models/bagel/bagel_transformer.py [hunk 1]
```diff
@@ -1690,11 +1697,15 @@ def generate_image(
...
-        # Lance / BAGEL both use this convention; without the +1 we silently
-        # run one fewer denoise iteration than the user asked for.
-        timesteps = torch.linspace(1, 0, num_timesteps + 1, device=x_t.device)
+        # Build the flow-matching schedule. Official BAGEL samples
+        # ``num_timesteps`` points over [1, 0] and drops the terminal t=0 (no
+        # dt), giving ``num_timesteps - 1`` Euler steps. Lance samples one extra
+        # point for ``num_timesteps`` steps. The ``_denoise_schedule_extra_step``
+        # class flag (overridden by ``LanceBagel``) selects the convention so
+        # that BAGEL stays bit-compatible with upstream ByteDance-Seed/BAGEL.
+        # See https://github.com/vllm-project/vllm-omni/issues/4470.
```
Comment (yuanheng-zhao) [inline vllm_omni/diffusion/models/bagel/bagel_transformer.py]: nit: can shorten a bit comments

**Review by yuanheng-zhao — APPROVED** [review]
LGTM

**Review by hsliuustc0106 — COMMENTED** [review]
Bagel denoising schedule fix looks correct.
